tq/basic_upload.go

```diff
@@ -74,6 +72,25 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 	}
 	defer f.Close()
 
+	if len(req.Header.Get("Content-Type")) == 0 {
+		buffer := make([]byte, 512)
+		n, err := f.Read(buffer)
+		if err != nil && err != io.EOF {
+			return errors.Wrap(err, "basic upload")
```

Comment:
I think this wrap description could be more informative. Perhaps something like:

```go
return errors.Wrap(err, "content type detect")
```